lib/msprime.c

```diff
@@ -1813,10 +1813,19 @@ static int MSP_WARN_UNUSED
 msp_store_edge(msp_t *self, double left, double right, tsk_id_t parent, tsk_id_t child)
 {
     int ret = 0;
-    tsk_edge_t *edge;
+    tsk_edge_t *edge, *last_edge;
     const double *node_time = self->tables->nodes.time;
 
     tsk_bug_assert(parent < (tsk_id_t) self->tables->nodes.num_rows);
+    if (self->num_buffered_edges > 0) {
+        last_edge = self->buffered_edges + self->num_buffered_edges - 1;
```

Comment:
```
tsk_edge_t *edge, last_edge;

last_edge = self->buffered_edges [self->num_buffered_edges - 1]
if (last_edge.parent != parent) {
```
is a bit more efficient (and obvious I think - I don't know why I was so fond of pointer addition tricks in the old days!)